drivers/SmartThings/harman-luxury/src/disco.lua

```diff
@@ -0,0 +1,130 @@
+local mdns = require "st.mdns"
+local socket = require "cosock.socket"
+local log = require "log"
+
+local disco_helper = require "disco_helper"
+local devices = require "devices"
+
+local SERVICE_TYPE = "_sue-st._tcp"
+local DOMAIN = "local"
+
+local Discovery = {}
```

Comment:
This one is a nit:

This file (and a few others) use a lot of string concatenation (the `..` operator).

`..` is fragile; it will crash on `nil` values if `tostring` isn't called first. Most of the usage here is well protected, so that's good.

However, the overall readability when there are multiple interleaved/concatenated variables in the middle of a string suffers, and the use of `tostring` to elide `nil` isn't very obvious.

In cases where one needs to insert multiple variables in to a string, it can often lead to much more readable and maintainble logging if `string.format()` is used instead of multiple concatenation operations.